### PR TITLE
ci: Fix flaky direct access transaction tests

### PR DESCRIPTION
**Pull request by dplewis**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

https://github.com/parse-community/parse-server/pull/9210 fixed the transaction flaky test for `batch.spec.js`. This PR fixes it for the ParseServerRESTConroller

https://github.com/parse-community/parse-server/actions/runs/10019940136/job/27697049679?pr=9218

### CODE REVIEW

File: spec/ParseServerRESTController.spec.js [hunk 1]
```diff
@@ -133,24 +134,11 @@ describe('ParseServerRESTController', () => {
     process.env.PARSE_SERVER_TEST_DB === 'postgres'
   ) {
     describe('transactions', () => {
-      beforeEach(async () => {
-        await TestUtils.destroyAllDataPermanently(true);
```
Comment (mtrezza) [inline spec/ParseServerRESTController.spec.js:137]: Is it important that `destroyAllDataPermanently` is not called anymore?

Comment (dplewis) [reply spec/ParseServerRESTController.spec.js:137]: No as destroyAllDataPermanently is called automatically after each test so it’s redundant here with the rest of the beforeEach

Comment (mtrezza) [reply spec/ParseServerRESTController.spec.js:137]: Good catch

**Review by mtrezza — APPROVED** [review]
That's an interesting fix; is it more reliable to reference a spy in a var `let createSpy = spyOn...` than to set the spy `spyOn(databaseAdapter, 'createObject').and.callThrough()` and then try to access it via the spied upon obj `databaseAdapter.createObject.calls`?

**Comment by dplewis** [comment]
Using `let createSpy = spyOn..` in a beforeEach guarantees the mock resets and allow you to programmatically reset the mocks for finer control of the tests.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0-alpha.7](https://github.com/parse-community/parse-server/releases/tag/7.3.0-alpha.7)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.3.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.3.0](https://github.com/parse-community/parse-server/releases/tag/7.3.0)

**Pull request merged**